### PR TITLE
[dashboard] Tell users prebuild is uploading

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Tell users that a workspace is uploading

## Related Issue(s)

Fixes #

## Release Notes

```release-note
```

### CODE REVIEW

File: components/dashboard/src/start/StartWorkspace.tsx [hunk 1]
```diff
@@ -340,7 +340,7 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
             return;
         }
 
-        if (workspaceInstance.status.phase === "building" || workspaceInstance.status.phase == "preparing") {
+        if (workspaceInstance.status.phase === "building" || workspaceInstance.status.phase === "preparing") {
```
Comment (geropl) [inline components/dashboard/src/start/StartWorkspace.tsx:343]: :+1:

**Review by geropl — APPROVED** [review]
:+1:

**Review by gtsiolis — APPROVED** [review]
Thanks, @svenefftinge! 🏀 

I could not test this in action due to some hiccups of the the preview environment, see [relevant discussion](https://gitpod.slack.com/archives/C032A46PWR0/p1652291101570529) (internal).

Left one suggestion to rephrase the title below based on our yesterday's discussion.

File: components/dashboard/src/start/StartWorkspace.tsx [hunk 2]
```diff
@@ -567,7 +567,12 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
             // Stopping means that the workspace is currently shutting down. It could go to stopped every moment.
             case "stopping":
                 if (isHeadless) {
-                    return <HeadlessWorkspaceView instanceId={this.state.workspaceInstance.id} />;
+                    return (
+                        <HeadlessWorkspaceView
+                            instanceId={this.state.workspaceInstance.id}
+                            title="Uploading the prebuild ..."
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/dashboard/src/start/StartWorkspace.tsx:573]: I find preparing and finalizing too abstract to bridge a few minutes of waiting. "Why does finalizing take so long?"

**Comment by svenefftinge** [comment]
Turns out this was the wrong place (CreateWorkspace is correct) and also we don't get updates on the prebuild.

File: components/dashboard/src/start/StartWorkspace.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gtsiolis) [reply components/dashboard/src/start/StartWorkspace.tsx:573]: The existing titles aim to abstract the sub-states of the loading cycle to make the loading friendlier to the user, that's why we're using _Checking_, _Creating_, & _Starting_ in the rest of the workspace start phases. We're currently hijacking the same title to show _Prebuild in Progress_ and do not group or relate the prebuild-in-progress phase with the rest of the workspace start phases.

Using the _Uploading_ term could suffice as a sub-state and also wouldn't hurt to also use it as a title as this is already a good improvement. 🤝  

| Uploading | Preparing |
|-|-|
| loading-1 | loading-2 |

**thought**: We could consider :a: reusing the same element from the prebuilds pages (see https://github.com/gitpod-io/gitpod/issues/9132) where we could surface these sub-states of the prebuild phase and :b: providing better visibility of the step sequence as currently it's not clear if the prebuild happens before or after the rest of the workspace start phases, etc. However, this is probably out of the scope of this PR. ➿ 

| BEFORE | AFTER (Early design) |
|-|-|
| loading-0 | loading-5 |

**Pull request closed without merging**